Add/gutenboarding login page part 2

roo2 opened the pull request:
#### Changes proposed in this Pull Request

Link login dialog into the gutenboarding flow.
Handle different states of the login page.
Remove `/gutenboarding/signup` and `gutenboarding/login/` routes which are now unused and don't work correctly when accessed directly

#### Planned but not included in this PR
* Handling 2FA users correctly
* Handling passwordles users correctly ([Issue link](https://github.com/Automattic/wp-calypso/issues/40004))

#### Testing instructions

Go through the `/gutenboarding` flow until you get to the *Create Site* button
Click *create site* to open the signup modal.
* There should be a login link
* The login link should take you to a login form ( in a way that's acceptably smooth )
NOTE: I've implemented this as two modals but do you think that the transition from one modal to the other is smooth enough or should I look to combine the two for a smoother transition
* Use your password manager to autofill the username and password (I tested 1password in FF so far)
* Fill in an incorrect username -> should show an error
* Fill in a correct username -> should clear the error and show the password field
* Fill in an incorrect password -> should show an error
* Click *Change username* link -> should make username editable again
* Enter correct username and password -> should now log you in, take you to the `create-site` route and on to edit your new site!

ramonjd commented on client/landing/gutenboarding/components/header/index.tsx:
```diff
@@ -116,9 +120,23 @@ const Header: FunctionComponent = () => {
 
 	useEffect( () => {
+		// make login a part of this
+		// we need to add something to determine if the user exists I think, and then just redirect.
```
Would `select( USER_STORE ).getCurrentUser()` do the trick?

roo2 replied:
oops those comments were obsolete!

ramonjd commented on client/landing/gutenboarding/components/login-form/index.tsx:
```diff
@@ -59,18 +85,32 @@ const LoginForm = () => {
+					disabled={ shouldShowPasswordField }
 					onChange={ setUsernameOrEmailVal }
 					placeholder={ NO_x(
 						'E.g., yourname@email.com',
```
Maybe we don't need a placeholder since the user can enter either a username or an email (?)

simison replied:
+1, I'd stick to how it's in production since I'm sure it was considered:

<img width="518" alt="image" src="https://user-images.githubusercontent.com/87168/76403633-db5c0180-638d-11ea-84b2-486d9dc6d11b.png">

simison replied:
Aside, if you decide to move on with email placeholder, we have internal example domains we can use. Examples across Calypso. :-)

roo2 replied:
ok, yep I'll take the placeholder out and make it the same as the existing login page 👍

ramonjd commented on client/landing/gutenboarding/components/login-form/index.tsx:
```diff
@@ -59,18 +85,32 @@ const LoginForm = () => {
+					label={ ! shouldShowPasswordField ? NO__( 'Email Address or Username' ) : '' }
 					value={ usernameOrEmailVal }
-					// disabled={ isLoading }
+					disabled={ shouldShowPasswordField }
```
I can't change my username if the password field is showing. Will this be in the final version?

roo2 replied:
that's the plan at the moment, it makes the implementation quite a bit easier/more elegant and matches what's in production now. on `/log-in`. I've updated the CSS so that the disabled state is now visible in chrome!

roo2 commented:
I've rebased this and Login now works!
I refactored this into components that match the login form's states. This pattern should support new states for `ENTER_2FA_CODE`, `WAITING_FOR_2FA_APP` etc nicely. 
I also tested with the 1password plugin in firefox on macos, it fills username and password successfully. I couldn't get the plugin to install correctly in chrome but I'll have another look at that later today/tomorrow and then look at how other password managers deal with it.
